cylc/flow/async_util.py

```diff
@@ -389,7 +389,7 @@ def _scandir(future, path, request):
     """Callback helper for scandir()."""
     future.set_result([
         Path(path, directory.name)
-        for directory in request.result
+        for directory in request.result or []
```

Comment:
```suggestion
        # request.result can be None (maybe because of an empty directory in cylc-run)
        for directory in request.result or []
```